frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/utils/conversionUtils.test.ts

```diff
@@ -0,0 +1,44 @@
+import type { Option } from 'app-shared/types/Option';
+import type { CodeListItem } from '@studio/components';
+import {
+  convertOptionsListToCodeListItemList,
+  convertCodeListItemListToOptionsList,
+} from './conversionUtils';
+
+// Test data:
+const optionsList: Option[] = [
+  { value: 'test', label: 'test label', description: 'description', helpText: 'help text' },
+];
+const codeListItemList: CodeListItem[] = [
+  { value: 'test', label: 'test label', description: 'description', helpText: 'help text' },
+];
```

Comment:
This is due to an issue i've thought about mentioning.
We are currently dealing with 2 types being duplicate of each other, `Option` <-> `CodeListItem`.
`Option` is a shared type while `CodeListItem` is a type used in the new Studio component `StudioCodeListEditor`.

I see 3 potential solutions to this problem:
1. Remove `CodeListItem` and stick with `Option`, this will result in the Studio component not knowing it's own types in it's folder structure as the type it refers to would be in the shared types folder, which is not a good thing.
2. Vise versa
3. Make `Option` inherit from CodeListItem, this way they can both exist and the change would not be too big. This would allow me to remove these convertion functions and typescript would be happy.

Which one should we go for?

I'm leaning towards the 3. option, until we have the meeting about Option vs. CodeList as our goto naming convention.

Reply:
Sounds like a solution! 😄 
But my first though was to export the type from StudioCodeList and use that in the whole solution? Cant wee how that is problematic? 🫣

Reply:
I'll refer to the meeting about `OptionList` vs `CodeList`
All options lists collected from our backend are `optionsLists`, so changing out all types of `Option[]` -> `CodeList` and `Option` -> `CodeListItem` would be against the conclusion we just came to.

I also see a problem with inheriting the types. Which type should inhert from which.

### Option Inherits from CodeListItem
This lets the StudioComponent have it's local type in it's entirety.
A downside is an increase in complexity when we want to change CodeListItem, we will have to Omit the new changes.

### CodeListItem inherits from Option
This lets us freely change CodeListItem once we decide to change it in the future.
A downside is we are now making the studio component reliant on the shared types and that will include a more difficult time with making Studio components a nuget package later. Erling mentioned this was a possibility and the main reason why all studio components should contain all required types and logic to be self sustained.

What do you think is the least bad option?

Reply:
first of all we need to keep studio-components independent and isolated. So the `CodeListItem` has to be unknown that there exists an `Option` type. Our implementation of the CodeListEditor in ux-editor needs to use `type Option = OptionListItem` and if we one day make changes to the CodeListItem we can omit the props we need in Option when that issue arises 😄

Reply:
:+1: ye, it was the solution i'm leaning towards